src/test/shell/bazel/target_compatible_with_test.sh

```diff
@@ -0,0 +1,696 @@
+#!/bin/bash
```

Comment:
This is a great set of tests. Could it be moved to src/test/shell/integration? That will cause it to be run more reliably on our internal codebase as well as for Bazel.

Reply:
Done.